caffeine@patapon.info/extension.js

```diff
@@ -191,67 +395,251 @@ class Caffeine extends PanelMenu.Button {
         }
         return inFullscreen;
     }
-
+    
     toggleFullscreen() {
+        this._manageScreenBlankState(false);
         Mainloop.timeout_add_seconds(2, () => {
-            if (this.inFullscreen && !this._apps.includes('fullscreen')) {
+            if (this.inFullscreen && !this._appInhibitedData.has('fullscreen')) {
                 this.addInhibit('fullscreen');
-                this._manageNightLight('disabled');
+                this._manageNightLight(false, false);
             }
         });
 
-        if (!this.inFullscreen && this._apps.includes('fullscreen')) {
+        if (!this.inFullscreen && this._appInhibitedData.has('fullscreen')) {
             this.removeInhibit('fullscreen');
-            this._manageNightLight('enabled');
+            this._manageNightLight(true, false);
         }
     }
 
     toggleState() {
+        this._manageScreenBlankState(false);
         if (this._state) {
-            this._apps.forEach(appId => this.removeInhibit(appId));
-            this._manageNightLight('enabled');
-        } else {
+            this._removeTimer(false);
+            this._appInhibitedData.forEach((data, appId) => 
+                this.removeInhibit(appId)
+            );
+            this._manageNightLight(true, false);            
+        } else {     
             this.addInhibit('user');
-            this._manageNightLight('disabled');
-        }
+            this._manageNightLight(false, false);
+        }        
     }
 
     addInhibit(appId) {
         this._sessionManager.InhibitRemote(appId,
-            0, 'Inhibit by %s'.format(IndicatorName), 12,
-            cookie => {
-                this._last_cookie = cookie;
-                this._last_app = appId;
+            0, 'Inhibit by %s'.format(IndicatorName), this.inhibitFlags,
+            cookie => {             
+                // Init app data
+                let data = {
+                    cookie: cookie,
+                    isToggled: true,
+                    isInhibited: false,
+                    object: '',
+                };                
+                this._appInhibitedData.set(appId, data);
             }
         );
+        this._inhibition_added_fifo.push(appId);
```

Comment:
Shouldn't this be done in the `InhibitRemote` callback ? So that we push the appID only if the call to `InhibitRemote` succeeds

Reply:
Indeed, make sense.